dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResultHydrated.java

```diff
@@ -0,0 +1,183 @@
+package com.dotcms.rendering.velocity.viewtools.navigation;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.velocity.tools.view.context.ViewContext;
+
+import com.liferay.portal.model.User;
+
+public final class NavResultHydrated extends NavResult{
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    final NavResult navResult;
+    final transient ViewContext context;
+
+    public NavResultHydrated(final NavResult navResult, final ViewContext context) {
+        super(navResult);
+        this.navResult = navResult;
+        this.context = context;
+
+    }
+
+
+
+    public boolean isActive() {
+        if (context != null && UtilMethods.isSet(navResult.getHref())) {
+            HttpServletRequest req = (HttpServletRequest) context.getRequest();
+            if (req != null) {
+                // We exclude the page name from the Request URI so we can check if page's parent
+                // object is the real active object
+                String reqURI = req.getRequestURI();
+                String parentPath = reqURI.substring(0, reqURI.lastIndexOf("/"));
+                if (!parentPath.endsWith("/"))
+                    // Adding a slash at the end of the path, so it avoids false positives
+                    // when two or more paths from the same level starts with the same name
+                    parentPath = parentPath + "/";
+                // If the current item is a folder, we check if it's part of current URI
+
+                // System.err.println(href + " : " + reqURI);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)